ejb3/src/test/java/org/jboss/as/ejb3/util/MdbValidationsUtilTest.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright (c) 2011, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.jboss.as.ejb3.util;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import org.jboss.jandex.ClassInfo;
+import org.jboss.jandex.DotName;
+import org.jboss.jandex.Index;
+import org.jboss.jandex.Indexer;
+import org.junit.Test;
+
+/**
+ *
+ * @author <a href="mailto:romain@redhat.com">Romain Pelisse</a>
+ */
+public class MdbValidationsUtilTest {
+
+    @Test
+    public void mdbWithFinalClass() throws IOException {
+        MdbValidationsUtil.assertMDBClassValidity(buildClassInfoForClass("org.jboss.as.ejb3.util.InvalidMdbFinalClass"));
```

Comment:
Why not use the type safe InvalidMdbFinalClass.class.getName() rather than a hard coded string?